pkg/util/winutil/pdhutil/pdhcounter.go

```diff
@@ -0,0 +1,181 @@
+		var sz uint32
+		sz = bufferSize
+		regerr := windows.RegQueryValueEx(syscall.HKEY_PERFORMANCE_DATA,
+			syscall.StringToUTF16Ptr("Counter 009"),
+			nil, // reserved
+			&regtype,
+			(*byte)(unsafe.Pointer(&counterlist[0])),
+			&sz)
+		if regerr == error(syscall.ERROR_MORE_DATA) {
+			// buffer's not big enough
+			bufferSize += bufferIncrement
+			continue
+		}
+		break
+	}
+	clist := winutil.ConvertWindowsStringList(counterlist)
+	for i := 0; i < len(clist); i += 2 {
+		ndx, _ := strconv.Atoi(clist[i])
+		counterToIndex[clist[i+1]] = ndx
+	}
+
+}
+
+// GetCounterSet returns an initialized PDH counter set.
+func GetCounterSet(className string, counterName string, instanceName string, verifyfn CounterInstanceVerify) (*PdhCounterSet, error) {
+	var p PdhCounterSet
+	p.countermap = make(map[string]win.PDH_HCOUNTER)
+	var ndx int
+	var err error
+	if ndx = getCounterIndex(className); ndx == -1 {
+		return nil, fmt.Errorf("Class name not found: %s", className)
+	}
+	p.className, err = pdhLookupPerfNameByIndex(ndx)
+	if err != nil {
+		return nil, err
+	}
+	if ndx = getCounterIndex(counterName); ndx == -1 {
+		return nil, fmt.Errorf("Class name not found: %s", counterName)
+	}
+	p.counterName, err = pdhLookupPerfNameByIndex(ndx)
+	if err != nil {
+		return nil, err
+	}
+	winerror := win.PdhOpenQuery(uintptr(0), uintptr(0), &p.query)
+	if win.ERROR_SUCCESS != winerror {
+		err = fmt.Errorf("Failed to open PDH query handle %d", winerror)
+		return nil, err
+	}
+	_, instances, err := pdhEnumObjectItems(p.className)
+	if err != nil {
+		return nil, err
+	}
+	if instanceName == "" && len(instances) > 0 {
+		// asked for all instances of this class
+		for _, inst := range instances {
+			if verifyfn != nil {
+				if verifyfn(inst) == false {
+					// verify function said not interested in this instance, move on
+					continue
+				}
+			}
+			path, err := pdhMakeCounterPath("", p.className, inst, p.counterName)
+			if err != nil {
+				continue
+			}
+			var hc win.PDH_HCOUNTER
+			winerror = win.PdhAddCounter(p.query, path, uintptr(0), &hc)
+			if win.ERROR_SUCCESS != winerror {
+				continue
+			}
+			p.countermap[inst] = hc
+		}
+	} else {
+		if instanceName != "" {
+			// they asked for specific instance
+			if len(instances) <= 0 {
+				return nil, fmt.Errorf("Requested instance of sigle instance counter")
+			}
+			found := false
+			for _, inst := range instances {
+				if inst == instanceName {
+					found = true
+					break
+				}
+			}
+			if !found {
+				return nil, fmt.Errorf("Didn't find instance name %s", instanceName)
+			}
+		}
+		path, err := pdhMakeCounterPath("", p.className, instanceName, p.counterName)
+		if err != nil {
+			return nil, err
+		}
+		winerror = win.PdhAddCounter(p.query, path, uintptr(0), &p.singleCounter)
+		if win.ERROR_SUCCESS != winerror {
+			return nil, fmt.Errorf("Failed to add single counter %d", winerror)
+		}
+	}
+	// do the initial collect now
+	win.PdhCollectQueryData(p.query)
+	return &p, nil
+
```

Comment:
sorry about this but can you align the return with the  bottom `}`?

I won't block on this, just a nit